nsxt/resource_nsxt_policy_segment_port.go

```diff
@@ -0,0 +1,223 @@
+		Importer: &schema.ResourceImporter{
+			State: getSegmentPortPathOrIDResourceImporter,
+		},
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"segment_path": {
+				Type:        schema.TypeString,
+				Description: "Path of the segment",
+				Required:    true,
+			},
+			"attachment": {
+				Type:        schema.TypeList,
+				Description: "VIF attachment",
+				Optional:    true,
+				Elem:        getPolicySegmentPortAttachmentSchema(),
+				MaxItems:    1,
+			},
+			"discovery_profile": {
+				Type:        schema.TypeList,
+				Description: "IP and MAC discovery profiles for this segment",
+				Elem:        getPolicySegmentDiscoveryProfilesSchema(),
+				Optional:    true,
+				MaxItems:    1,
+			},
+			"qos_profile": {
+				Type:        schema.TypeList,
+				Description: "QoS profiles for this segment",
+				Elem:        getPolicySegmentQosProfilesSchema(),
+				Optional:    true,
+				MaxItems:    1,
+			},
+			"security_profile": {
+				Type:        schema.TypeList,
+				Description: "Security profiles for this segment",
+				Elem:        getPolicySegmentSecurityProfilesSchema(),
+				Optional:    true,
+				MaxItems:    1,
+			},
+		},
+	}
+}
+
+var allocateAddresses = []string{"IP_POOL", "MAC_POOL", "BOTH", "DHCP", "DHCPV6", "SLAAC", "NONE"}
+
+func getPolicySegmentPortAttachmentSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"allocate_addresses": {
+				Type:         schema.TypeString,
+				Description:  "Indicate how IP will be allocated for the port. Allowed values are IP_POOL, MAC_POOL, BOTH, DHCP, DHCPV6, SLAAC, NONE",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(allocateAddresses, false),
+			},
+			"app_id": {
+				Type:        schema.TypeString,
+				Description: "ID used to identify/look up a child attachment behind a parent attachment",
+				Optional:    true,
+			},
+			"evpn_vlans": {
+				Type:        schema.TypeList,
+				Description: "Evpn tenant VLAN IDs the Parent logical-port serves.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				MaxItems: 1000,
+			},
+			"hyperbus_mode": {
+				Type:         schema.TypeString,
+				Description:  "ID used to identify/look up a child attachment behind a parent attachment",
+				Optional:     true,
+				Default:      "DISABLE",
+				ValidateFunc: validation.StringInSlice([]string{"ENABLE", "DISABLE"}, false),
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Description:  "Type of port attachment. PARENT type is automatically set if evpn_vlans or hyperbus_mode is configured. INDEPENDENT type is automatically set for ports that belong to Segment of type DVPortgroup. STATIC type is deprecated.",
```

Comment:
Probably this one should be computed too, since in some cases is set by the backend.